File: Communication entre processus/exo4.1.py
import os
import sys

def main():
    r, w = os.pipe()

    pid = os.fork()

    if pid == 0:
        os.close(w)
        
        os.environ["MON_TUBE"] = str(r)
        
        path_to_python = sys.executable
        script_fils = "./fils_env.py"
        
        os.execv(path_to_python, [path_to_python, script_fils])
        os._exit(1)

    else:
        os.close(r)
        
        msg = "Message via environnement"
        os.write(w, (msg + '\0').encode())
        
        os.close(w)
        os.wait()

if __name__ == "__main__":
    main()

File: Communication entre processus/exo4.1.c
#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>
#include <string.h>

int main() 
{
    int p[2];
    pipe(p);

    if (fork() == 0) 
    {
        char fd_str[10]; 
        sprintf(fd_str, "%d", p[0]);
        
        
        setenv("MON_TUBE", fd_str, 1);
        
        char *args[] = {"./fils_env", NULL};
        execv(args[0], args);
        exit(1);
    } 
    else 
    {
        close(p[0]);
        char *msg = "Message via environnement";
        write(p[1], msg, strlen(msg) + 1);
        close(p[1]);
        wait(NULL);
    }
    return 0;
}